[Bug] Build error under gcc/g++ 13 + Ubuntu24.04.2

### Apache Cloudberry version

main

### What happened

```
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CDatumGenericGPDB.o CDatumGenericGPDB.cpp
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamString.h:14,
                 from COstreamString.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from COstreamString.cpp:15:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/error/CLogger.h:15,
                 from CLogger.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from CLogger.cpp:20:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../src/backend/gporca/libgpdbcost/include/gpdbcost/CCostModelParamsGPDB.h:16,
                 from CCostModelParamsGPDB.cpp:12:
../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../src/backend/gporca/libgpdbcost/include/gpdbcost/CCostModelParamsGPDB.h:17:
../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CHashMap.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatum.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatumBool.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/CDatumBoolGPDB.h:16,
                 from CDatumBoolGPDB.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatum.h:18:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CList.h:21,
                 from ../../../../../../src/backend/gporca/libgpopt/include/gpopt/base/CColumnFactory.h:15,
                 from ../../../../../../src/backend/gporca/libgpopt/include/gpopt/base/CAutoOptCtxt.h:17,
                 from CAutoOptCtxt.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpopt/include/gpopt/metadata/CName.h:17,
                 from ../../../../../../src/backend/gporca/libgpopt/include/gpopt/base/CColRef.h:20,
                 from ../../../../../../src/backend/gporca/libgpopt/include/gpopt/base/CColRefSet.h:17,
                 from ../../../../../../src/backend/gporca/libgpopt/include/gpopt/base/CColumnFactory.h:18:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../src/backend/gporca/libgpdbcost/include/gpdbcost/CCostModelParamsGPDB.h:16,
                 from ../../../../../src/backend/gporca/libgpdbcost/include/gpdbcost/CCostModelGPDB.h:17,
                 from CCostModelGPDB.cpp:12:
../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../src/backend/gporca/libgpdbcost/include/gpdbcost/CCostModelParamsGPDB.h:17:
../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: COstreamString.o] Error 1
make[6]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpos/src/io'
make[5]: *** [../../../../../src/backend/common.mk:39: io-recursive] Error 2
make[5]: *** Waiting for unfinished jobs....
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CDatumInt2GPDB.o CDatumInt2GPDB.cpp
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CHashMap.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatum.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatumGeneric.h:17,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/CDatumGenericGPDB.h:16,
                 from CDatumGenericGPDB.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatum.h:18:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CLogger.o] Error 1
make[6]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpos/src/error'
make[5]: *** [../../../../../src/backend/common.mk:39: error-recursive] Error 2
make[5]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpos/src'
make[4]: *** [../../../../src/backend/common.mk:39: src-recursive] Error 2
make[4]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpos'
make[3]: *** [../../../src/backend/common.mk:39: libgpos-recursive] Error 2
make -C md all
make[3]: *** Waiting for unfinished jobs....
make[6]: Entering directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src/md'
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CDXLBucket.o CDXLBucket.cpp
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CAutoOptCtxt.o] Error 1
make[6]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpopt/src/base'
make[5]: *** [../../../../../src/backend/common.mk:39: base-recursive] Error 2
make[5]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpopt/src'
make[4]: *** [../../../../src/backend/common.mk:39: src-recursive] Error 2
make[4]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpopt'
make[3]: *** [../../../src/backend/common.mk:39: libgpopt-recursive] Error 2
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../src/backend/gporca/libnaucrates/include -I../../../../../src/backend/gporca/libgpopt/include -I../../../../../src/backend/gporca/libgpos/include -I../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o ICostModel.o ICostModel.cpp
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDatumBoolGPDB.o] Error 1
make -C operators all
make[6]: *** Waiting for unfinished jobs....
make[6]: Entering directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src/operators'
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CDXLColDescr.o CDXLColDescr.cpp
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CHashMap.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatum.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatumInt2.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/CDatumInt2GPDB.h:16,
                 from CDatumInt2GPDB.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/base/IDatum.h:18:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:19,
                 from CDXLBucket.cpp:13:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:21:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:14,
                 from ICostModel.cpp:13:
../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDatumGenericGPDB.o] Error 1
make -C parser all
make[6]: Entering directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src/parser'
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CParseHandlerAgg.o CParseHandlerAgg.cpp
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDXLBucket.o] Error 1
make[6]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src/md'
make[5]: *** [../../../../../src/backend/common.mk:39: md-recursive] Error 2
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CDXLColRef.o CDXLColRef.cpp
make[5]: *** Waiting for unfinished jobs....
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLColDescr.h:18,
                 from CDXLColDescr.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CMDIdGPDB.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLColDescr.h:20:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[5]: *** [<builtin>: ICostModel.o] Error 1
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CDXLCtasStorageOptions.o CDXLCtasStorageOptions.cpp
make[5]: *** Waiting for unfinished jobs....
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDatumInt2GPDB.o] Error 1
make[6]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src/base'
make[5]: *** [../../../../../src/backend/common.mk:39: base-recursive] Error 2
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CDXLDatum.o CDXLDatum.cpp
touch objfiles.txt
make[3]: Leaving directory '/home/gpadmin/cloudberry/src/backend/libpq'
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CParseHandlerAppend.o CParseHandlerAppend.cpp
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDXLColDescr.o] Error 1
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CParseHandlerArray.o CParseHandlerArray.cpp
make[6]: *** Waiting for unfinished jobs....
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLNode.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalAgg.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/parser/CParseHandlerAgg.h:17,
                 from CParseHandlerAgg.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:21,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedColumn.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedRelation.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLProperties.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLNode.h:19:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLColRef.h:18,
                 from CDXLColRef.cpp:13:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CMDName.h:17,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLColRef.h:21:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLCtasStorageOptions.h:18,
                 from CDXLCtasStorageOptions.cpp:13:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CMDName.h:17,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLCtasStorageOptions.h:21:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:18,
                 from CDXLDatum.cpp:18:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CBitSet.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalAppend.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/parser/CParseHandlerAppend.h:17,
                 from CParseHandlerAppend.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:21,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedColumn.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedRelation.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLOperator.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysical.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalAppend.h:21:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDXLColRef.o] Error 1
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CParseHandlerAssert.o CParseHandlerAssert.cpp
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDXLCtasStorageOptions.o] Error 1
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CParseHandlerBase.o CParseHandlerBase.cpp
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLNode.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalAgg.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/parser/CParseHandlerArray.h:17,
                 from CParseHandlerArray.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:21,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedColumn.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedRelation.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLProperties.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLNode.h:19:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CDXLDatum.o] Error 1
make[6]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src/operators'
make[5]: *** [../../../../../src/backend/common.mk:39: operators-recursive] Error 2
g++ -std=c++14 -Werror -Wextra -Wpedantic -fno-omit-frame-pointer -Wall -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wimplicit-fallthrough=3 -Wcast-function-type -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fexcess-precision=standard -g -O3 -fPIC  -I../../../../../../src/backend/gporca/libgpdbcost/include -I../../../../../../src/backend/gporca/libnaucrates/include -I../../../../../../src/backend/gporca/libgpopt/include -I../../../../../../src/backend/gporca/libgpos/include -I../../../../../../src/include   -D_GNU_SOURCE -I/usr/include/libxml2  -I/usr/include/xercesc  -c -o CParseHandlerBroadcastMotion.o CParseHandlerBroadcastMotion.cpp
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CParseHandlerAgg.o] Error 1
make[6]: *** Waiting for unfinished jobs....
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CParseHandlerAppend.o] Error 1
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLOperator.h:17,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysical.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalAssert.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/parser/CParseHandlerAssert.h:17,
                 from CParseHandlerAssert.cpp:13:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:21,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedColumn.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedRelation.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLOperator.h:19:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CBitSet.h:16,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CEnumSet.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/CDXLUtils.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/parser/CParseHandlerBase.h:19,
                 from CParseHandlerBase.cpp:13:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:21,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedColumn.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedRelation.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLProperties.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLNode.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/CDXLUtils.h:24:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CParseHandlerArray.o] Error 1
In file included from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWString.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringStatic.h:15,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstream.h:17,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/io/COstreamBasic.h:14,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/utils.h:19,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CRefCount.h:22,
                 from ../../../../../../src/backend/gporca/libgpos/include/gpos/common/CDynamicPtrArray.h:15,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalMotion.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalBroadcastMotion.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/parser/CParseHandlerBroadcastMotion.h:17,
                 from CParseHandlerBroadcastMotion.cpp:12:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringBase.h:79:22: error: ‘virtual gpos::BOOL gpos::CWStringBase::Equals(const gpos::WCHAR*) const’ was hidden [-Werror=overloaded-virtual=]
   79 |         virtual BOOL Equals(const WCHAR *str) const;
      |                      ^~~~~~
In file included from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLDatum.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLBucket.h:21,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedColumn.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/md/CDXLStatsDerivedRelation.h:18,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLProperties.h:16,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLNode.h:19,
                 from ../../../../../../src/backend/gporca/libnaucrates/include/naucrates/dxl/operators/CDXLPhysicalMotion.h:20:
../../../../../../src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h:59:14: note:   by ‘gpos::CWStringConst::Equals’
   59 |         BOOL Equals(const CWStringBase *str) const override;
      |              ^~~~~~
cc1plus: all warnings being treated as errors
make[5]: *** [<builtin>: CCostModelParamsGPDB.o] Error 1
cc1plus: all warnings being treated as errors
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CParseHandlerBase.o] Error 1
make[6]: *** [<builtin>: CParseHandlerAssert.o] Error 1
cc1plus: all warnings being treated as errors
make[6]: *** [<builtin>: CParseHandlerBroadcastMotion.o] Error 1
make[6]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src/parser'
make[5]: *** [../../../../../src/backend/common.mk:39: parser-recursive] Error 2
make[5]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates/src'
make[4]: *** [../../../../src/backend/common.mk:39: src-recursive] Error 2
make[4]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libnaucrates'
make[3]: *** [../../../src/backend/common.mk:39: libnaucrates-recursive] Error 2
cc1plus: all warnings being treated as errors
make[5]: *** [<builtin>: CCostModelGPDB.o] Error 1
make[5]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpdbcost/src'
make[4]: *** [../../../../src/backend/common.mk:39: src-recursive] Error 2
make[4]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca/libgpdbcost'
make[3]: *** [../../../src/backend/common.mk:39: libgpdbcost-recursive] Error 2
make[3]: Leaving directory '/home/gpadmin/cloudberry/src/backend/gporca'
make[2]: *** [common.mk:39: gporca-recursive] Error 2
make[2]: Leaving directory '/home/gpadmin/cloudberry/src/backend'
make[1]: *** [Makefile:45: all-backend-recurse] Error 2
make[1]: Leaving directory '/home/gpadmin/cloudberry/src'
make: *** [GNUmakefile:11: all-src-recurse] Error 2
make: Leaving directory '/home/gpadmin/cloudberry'
```

### What you think should happen instead

_No response_

### How to reproduce

https://cloudberry.apache.org/docs/deployment/quick-build

Run `make -j$(nproc) -C ~/cloudberry`, and the errors will return.

### Operating System

Ubuntu 24.04.2 LTS, gcc/g++ 13.3.0

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/apache/cloudberry/blob/main/CODE_OF_CONDUCT.md).

--- a/contrib/pax_storage/src/cpp/storage/micro_partition.cc
+++ b/contrib/pax_storage/src/cpp/storage/micro_partition.cc
@@ -91,7 +91,7 @@ size_t MicroPartitionReaderProxy::GetTupleCountsInGroup(size_t group_index) {
 
 std::unique_ptr<ColumnStatsProvider>
 MicroPartitionReaderProxy::GetGroupStatsInfo(size_t group_index) {
-  return std::move(reader_->GetGroupStatsInfo(group_index));
+  return reader_->GetGroupStatsInfo(group_index);
 }
 
 std::unique_ptr<MicroPartitionReader::Group>

--- a/src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h
+++ b/src/backend/gporca/libgpos/include/gpos/string/CWStringConst.h
@@ -57,6 +57,9 @@ public:
 
 	// checks whether the string is byte-wise equal to another string
 	BOOL Equals(const CWStringBase *str) const override;
+
+	// bring base class Equals(const WCHAR*) into scope to avoid hiding
+	using CWStringBase::Equals;
 };
 }  // namespace gpos
 

